pkg/metrics/metrics.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// NfsProvisionerSubsystem is prometheus subsystem name.
+	NfsVolumeProvisionerSubsystem = "nfs_volume_provisioner"
+
+	// Metrics
+	// ProvisionerRequestCreate represents metrics related to create resource request.
+	ProvisionerRequestCreate = "create"
+	// ProvisionerRequestDelete represents metrics related to delete resource request.
+	ProvisionerRequestDelete = "delete"
+
+	// Labels
+	Process = "process"
+)
+
+var (
+	// PersistentVolumeDeleteTotal is used to collect accumulated count of persistent volumes deleted.
+	PersistentVolumeDeleteTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: NfsVolumeProvisionerSubsystem,
+			Name:      "persistentvolume_delete_total",
+			Help:      "Total number of persistent volumes deleted",
```

Comment:
"Total" here refers to the total number of volumes deleted - by this provisioner pod. 

I can think of one usecase for these metrics - to check on the health of the provisioner to see if provisioner is able to successfully create volume or if the failures are increasing. 

Are there other usecase for these metrics? 

Another interesting metric can be to expose the average duration taken for creation or deletion of the volumes. Probably these can be covered in CSI. 

Overall good to merge after getting a response to the usecase query.

Reply:
It will help to check stale pv count also from `persistentvolume_delete_failed_total`